selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/NewWorkspacePageTest.java

```diff
@@ -0,0 +1,550 @@
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.Workspaces;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.TimeoutException;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/** @author Ihor Okhrimenko */
+public class NewWorkspacePageTest {
+  private static final int EXPECTED_QUICK_START_STACKS_COUNT = 15;
+  private static final int EXPECTED_SINGLE_MACHINE_STACKS_COUNT = 34;
+  private static final int EXPECTED_ALL_STACKS_COUNT = 39;
+  private static final int EXPECTED_MULTI_MACHINE_STACKS_COUNT = 5;
+  private static final String EXPECTED_WORKSPACE_NAME_PREFIX = "wksp-";
+  private static final String MACHINE_NAME = "dev-machine";
+  private static final double MAX_RAM_VALUE = 100.0;
+  private static final double MIN_RAM_VALUE = 0.5;
+  private static final double RAM_CHANGE_STEP = 0.5;
+  private static final String NAME_WITH_ONE_HUNDRED_SYMBOLS =
+      "wksp-ppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppppp";
+  private static final List<String> NOT_VALID_NAMES =
+      asList("wksp-", "-wksp", "wk sp", "wk_sp", "wksp@", "wksp$", "wksp&", "wksp*");
+  private static final String LETTER_FOR_SEARCHING = "j";
+  private static final List<String> EXPECTED_JDK_STACKS = asList("java-default", "che-in-che");
+  private static List<String> EXPECTED_QUICK_START_STACKS =
+      asList(
+          "blank-default",
+          "java-default",
+          "dotnet-default",
+          "android-default",
+          "cpp-default",
+          "che-in-che",
+          "go-default",
+          "java-theia-openshift",
+          "node-default",
+          "openshift-default",
+          "openshift-sql",
+          "php-default",
+          "platformio",
+          "python-default",
+          "rails-default");
+
+  private static List<String> EXPECTED_SINGLE_MACHINE_STACKS =
+      asList(
+          "blank-default",
+          "java-default",
+          "dotnet-default",
+          "android-default",
+          "cpp-default",
+          "centos",
+          "centos-go",
+          "nodejs4",
+          "wildfly-swarm",
+          "ceylon-java-javascript-dart-centos",
+          "debian",
+          "debianlsp",
+          "che-in-che",
+          "vert.x",
+          "go-default",
+          "hadoop-default",
+          "java-centos",
+          "java-debian",
+          "kotlin-default",
+          "node-default",
+          "openshift-default",
+          "php-default",
+          "php-gae",
+          "php5.6-default",
+          "platformio",
+          "python-default",
+          "python-2.7",
+          "python-gae",
+          "rails-default",
+          "selenium",
+          "spring-boot",
+          "tomee-default",
+          "ubuntu",
+          "zend");
+
+  private static final List<String> EXPECTED_ALL_STACKS =
+      Arrays.asList(
```

Comment:
May be make sense put long lists into resources for making the code more shorter?

Reply:
In my opinion, in this case, of course, it makes a code shorter, but more complicated.